cmd/bbolt/main.go

```diff
@@ -1353,7 +1419,37 @@ func (cmd *benchCommand) runReadsSequential(db *bolt.DB, options *BenchOptions,
 				numReads++
 				results.AddCompletedOps(1)
 				if v == nil {
-					return errors.New("invalid value")
+					return ErrInvalidValue
+				}
+			}
+
+			if options.WriteMode == "seq" && numReads != options.Iterations {
+				return fmt.Errorf("read seq: iter mismatch: expected %d, got %d", options.Iterations, numReads)
+			}
+
+			// Make sure we do this for at least a second.
+			if time.Since(t) >= time.Second {
+				break
+			}
+		}
+
+		return nil
+	})
+}
+
+func (cmd *benchCommand) runReadsRandom(db *bolt.DB, options *BenchOptions, keys [][]byte, results *BenchResults) error {
+	return db.View(func(tx *bolt.Tx) error {
+		t := time.Now()
+
+		for {
+			numReads := int64(0)
+			b := tx.Bucket(benchBucketName)
+			for _, key := range keys {
+				v := b.Get(key)
+				numReads++
+				results.AddCompletedOps(1)
+				if v == nil {
+					return ErrInvalidValue
```

Comment:
Should we add similar check (see below) for `runReadsRandom`?

```
if options.WriteMode == "seq-nest" && numReads != options.Iterations {
    return fmt.Errorf("read seq-nest: iter mismatch: expected %d, got %d", options.Iterations, numReads)
}

// Make sure we do this for at least a second.
if time.Since(t) >= time.Second {
    break
}
```